zcash_client_backend/src/proto.rs

```diff
@@ -359,72 +359,84 @@ impl proposal::Proposal {
-                    txid: rec_note.txid().as_ref().to_vec(),
-                    value_pool: proposal::ValuePool::from(rec_note.note().protocol()).into(),
-                    index: rec_note.output_index().into(),
-                    value: rec_note.note().value().into(),
-                })
-            }))
-            .collect();
+                let inputs = step
+                    .transparent_inputs()
+                    .iter()
+                    .map(|utxo| proposal::ProposedInput {
+                        txid: utxo.outpoint().hash().to_vec(),
+                        value_pool: proposal::ValuePool::Transparent.into(),
+                        index: utxo.outpoint().n(),
+                        value: utxo.txout().value.into(),
+                    })
+                    .chain(step.shielded_inputs().iter().flat_map(|s_in| {
+                        s_in.notes().iter().map(|rec_note| proposal::ProposedInput {
+                            txid: rec_note.txid().as_ref().to_vec(),
+                            value_pool: proposal::ValuePool::from(rec_note.note().protocol())
+                                .into(),
+                            index: rec_note.output_index().into(),
+                            value: rec_note.note().value().into(),
+                        })
+                    }))
+                    .collect();
 
-        let payment_output_pools = value
-            .payment_pools()
-            .iter()
-            .map(|(idx, pool_type)| proposal::PaymentOutputPool {
-                payment_index: u32::try_from(*idx).expect("Payment index fits into a u32"),
-                value_pool: proposal::ValuePool::from(*pool_type).into(),
+                let payment_output_pools = step
+                    .payment_pools()
+                    .iter()
+                    .map(|(idx, pool_type)| proposal::PaymentOutputPool {
+                        payment_index: u32::try_from(*idx).expect("Payment index fits into a u32"),
+                        value_pool: proposal::ValuePool::from(*pool_type).into(),
+                    })
+                    .collect();
+
+                let balance = Some(proposal::TransactionBalance {
+                    proposed_change: step
+                        .balance()
+                        .proposed_change()
+                        .iter()
+                        .map(|change| proposal::ChangeValue {
+                            value: change.value().into(),
+                            value_pool: proposal::ValuePool::from(change.output_pool()).into(),
+                            memo: change.memo().map(|memo_bytes| proposal::MemoBytes {
+                                value: memo_bytes.as_slice().to_vec(),
+                            }),
+                        })
+                        .collect(),
+                    fee_required: step.balance().fee_required().into(),
+                });
+
+                #[allow(deprecated)]
```

Comment:
Is this annotation needed on `ProposalStep`? I think it was only here because of the non-ZIP 317 fee rules, which are not on `ProposalStep`.